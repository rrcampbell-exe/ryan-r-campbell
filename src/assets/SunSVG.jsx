import React from 'react'

const SunSVG = () => {
  return (
    <svg className="banner-svg" id="sun-svg" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
      <circle cx="12" cy="12" r="6" stroke-width="2.5"/>
      <path d="M12 2V3" stroke-width="2.5" stroke-linecap="round"/>
      <path d="M12 21V22" stroke-width="2.5" stroke-linecap="round"/>
      <path d="M22 12L21 12" stroke-width="2.5" stroke-linecap="round"/>
      <path d="M3 12L2 12" stroke-width="2.5" stroke-linecap="round"/>
      <path d="M19.0708 4.92969L18.678 5.32252" stroke-width="2.5" stroke-linecap="round"/>
      <path d="M5.32178 18.6777L4.92894 19.0706" stroke-width="2.5" stroke-linecap="round"/>
      <path d="M19.0708 19.0703L18.678 18.6775" stroke-width="2.5" stroke-linecap="round"/>
      <path d="M5.32178 5.32227L4.92894 4.92943" stroke-width="2.5" stroke-linecap="round"/>
      </svg>
  )
}

export default SunSVG
