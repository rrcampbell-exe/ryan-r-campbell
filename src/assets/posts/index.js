import legacyPosts from './legacy-posts'
import posts from './new-posts'

export { legacyPosts, posts }
