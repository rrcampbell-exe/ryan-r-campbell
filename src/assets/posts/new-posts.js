const posts = [
  {
    id: 10090,
    date: "2024-05-24T06:00:00",
    slug: "let-there-be-light-dark",
    link: "https://ryanrcampbell.com/2024/05/24/let-there-be-light-dark",
    title: {
      rendered: "Let There Be light-dark()"
    },
    content: {
      rendered:
        '',
      protected: false
    },
    excerpt: {
      rendered: "In the beginning, there was light mode, but lo! After a time, the retinas of webizens did cry out, yearning for reprieve.",
      protected: false
    },
    tags: ['CSS', 'Web Development', 'Site Migration'],
    episode_featured_image: "/post-content/2024/05/let-there-be-light-dark-cover-img.png"
  },
  {
    id: 10089,
    date: "2024-05-23T06:00:00",
    slug: "book-review-the-good-enough-job",
    link: "https://ryanrcampbell.com/2024/05/23/book-review-the-good-enough-job",
    title: {
      rendered: "Book Review: The Good Enough Job"
    },
    content: {
      rendered:
        '',
      protected: false
    },
    excerpt: {
      rendered: "The right book. The right time. Right on.",
      protected: false
    },
    tags: ['Book Review', 'Personal Development', 'Identity'],
    episode_featured_image: "/post-content/2024/05/book-review-the-good-enough-job-cover-img.png",
    featured_image_alt: "The book cover for The Good Enough Job rests on a white table alongside a green succulent plant."
  },
  {
    id: 10088,
    date: "2024-05-03T06:00:00",
    slug: "welcome-to-the-new-ryan-r-campbell",
    link: "https://ryanrcampbell.com/2024/05/03/welcome-to-the-new-ryan-r-campbell",
    title: {
      rendered: "Welcome to the New Ryan R. Campbell (Or at Least the New Website)"
    },
    content: {
      rendered:
        '',
      protected: false
    },
    excerpt: {
      rendered: "It's been a while—nearly two years, to be exact—but golly, do I have some updates.",
      protected: false
    },
    tags: ['Site Migration', 'Web Development', 'On Writing'],
    episode_featured_image: "/post-content/2024/05/welcome-to-the-new-site-cover-img.png",
    featured_image_alt: "A cut-out image of Ryan R. Campbell next to text that welcomes visitors to the new website."
  }
];


export default posts
