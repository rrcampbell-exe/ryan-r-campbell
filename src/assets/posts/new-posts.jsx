const posts = [
  {
    id: 10088,
    date: "2024-05-03T06:00:00",
    slug: "welcome-to-the-new-ryan-r-campbell",
    link: "https://ryanrcampbell.com/2024/05/03/welcome-to-the-new-ryan-r-campbell",
    title: {
      rendered: "Welcome to the New Ryan R. Campbell (Or at Least the New Website)"
    },
    content: {
      rendered:
        `<div>
          <div class='Text-And-Image-Container'>
            <div class='text-block'>
              <h2>Well, I'm Back</h2>
              <p>It's been a while—nearly two years, to be exact—but golly, do I have some updates, one of which is the very site you're visiting now.</p>
              <p>But first, since most of you are more likely familiar with me as an author than anything else (at least at this point), let's start there.</p>
              <h2>Author Business</h2>
              <p>Hey, remember <em><a href='/books'>Scambait?</a></em> When I last posted, it had hardly been out for a month. Since then, your support and reviews have been wonderful to be on the receiving end of, and I'm thrilled to announce that (back in 2022), <em><b>Scambait</em> was named a finalist in the International Book Awards!</b></p>
              <p>Of books I've written, <em>Scambait</em> remains a favorite for me, personally, so it was deeply gratifying to see it get that recognition.</p>
              <p>In light of <em>Scambait</em>'s success, one might be wondering, "what's next?" or, given that it's been two years without a writerly update, "what happened?"</p>
              <p>Be patient. We're getting to that. Like, now.</p>
            </div>
            <div class='image-block'>
              <img src='/books/scambait/scambait-flat-cover-international-book-awards-readers-favorite-sticker.png' />
              <div class='caption-container'>
                <span>The 2022 International Book Awards finalist in Humor & Satire</span>
              </div>
            </div>
          </div>
          <div>
            <h2>Business Business</h2>
            <p>Those of you who <a href='http://eepurl.com/dhBxcv' rel='noopener noreferrer' target='_blank'>subscribe my newsletter</a>, follow me on Twitch (or did back when I was still streaming), or otherwise know me from around the web may also recall that in late 2021, my wife and I moved from Madison to Milwaukee. You might also know this move was part of a career transition for me.</p>
            <p>For those who want to learn more, I documented all of that change extensively in the extended version of this site's <a href='/about'>new about page</a>, but one of the consequences of adapting to my new life as a full-time software engineer was that writing fell to the wayside.</p>
            <p>Like way-way-way to the wayside.</p>
            <div class='Quote'>
              <div>
                <svg class='quotes-svg' viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg">
                  <g>
                    <path fill="none" d="M0 0h24v24H0z"/>
                    <path d="M4.583 17.321C3.553 16.227 3 15 3 13.011c0-3.5 2.457-6.637 6.03-8.188l.893 1.378c-3.335 1.804-3.987 4.145-4.247 5.621.537-.278 1.24-.375 1.929-.311 1.804.167 3.226 1.648 3.226 3.489a3.5 3.5 0 0 1-3.5 3.5c-1.073 0-2.099-.49-2.748-1.179zm10 0C13.553 16.227 13 15 13 13.011c0-3.5 2.457-6.637 6.03-8.188l.893 1.378c-3.335 1.804-3.987 4.145-4.247 5.621.537-.278 1.24-.375 1.929-.311 1.804.167 3.226 1.648 3.226 3.489a3.5 3.5 0 0 1-3.5 3.5c-1.073 0-2.099-.49-2.748-1.179z"/>
                  </g>
                </svg>
                <h3><em>... writing fell to the wayside. Like way-way-way to the wayside.</em></h3>
              </div>
            </div>
            <p>It pains me to share that since my last post in mid-May of 2022, I've sketched out ideas for novels, short stories, and even a poem or two, but when it comes to writing words in an actual manuscript, well, I have nothing to report.</p>
            <p>This is likely astonishing for those of you who recall I was at one time a strong advocate for <a href='/2017/08/25/finding-a-writing-routine'>writing every day</a> (or at least as close to every day as is possible), but at some point after <a href='/2022/05/19/from-conference-to-coffee-on-the-beach/'>that 2022 vacation to Puerto Rico</a>, I realized I could no longer sustain that intense writing regimen while remaining sufficiently focused on the new career I'd uprooted my life for. Simply put, something had to give.</p>
            <p>But it's not all bad news (I'll encourage you again to read the extended version of the story on the <a href='/about'>new about page</a> to better understand everything that went into making the move to Milwaukee a successful one).</p>
            <p>In fact, I'd say the break was a necessary one, and one that has me ready, again, to get back into the groove. A <em>new</em> groove.</p>
            <p>Please don't throw it off.</p>
          </div>
          <div>
            <h2>New Business</h2>
            <p>In my time away, I realized something: at one point, I'd become so caught up in the idea of being an author that I forsook so many other parts of myself in service of that singular identity. And that was, dare I say, not great.</p>
            <p>Like I detail throughout this new site and over on that <a href='/about'>new about page</a> (can you tell I want you to read it?), I've returned to embracing the idea of self as <em>more than</em>, by which I mean we are more than any one thing, idea, or act. We are <em>everything people</em>, and I felt it was time I start acting like it.</p>
            <p>Inspired by this notion to once again to clickity-clack at my keyboard, I decided enough was enough: I was no longer going to pay WordPress for something that, equipped with my new skills as a software engineer and web developer, I could do on my own.</p>
            <p>And so, this new site was born. From scratch. Bits and bytes HTML tags and all.</p>
          </div>
          <div class='Quote'>
          <div>
            <svg class='quotes-svg' viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg">
              <g>
                <path fill="none" d="M0 0h24v24H0z"/>
                <path d="M4.583 17.321C3.553 16.227 3 15 3 13.011c0-3.5 2.457-6.637 6.03-8.188l.893 1.378c-3.335 1.804-3.987 4.145-4.247 5.621.537-.278 1.24-.375 1.929-.311 1.804.167 3.226 1.648 3.226 3.489a3.5 3.5 0 0 1-3.5 3.5c-1.073 0-2.099-.49-2.748-1.179zm10 0C13.553 16.227 13 15 13 13.011c0-3.5 2.457-6.637 6.03-8.188l.893 1.378c-3.335 1.804-3.987 4.145-4.247 5.621.537-.278 1.24-.375 1.929-.311 1.804.167 3.226 1.648 3.226 3.489a3.5 3.5 0 0 1-3.5 3.5c-1.073 0-2.099-.49-2.748-1.179z"/>
              </g>
            </svg>
            <h3><em>I'd become so caught up in the idea of a singular identity, and that was... not great.</em></h3>
          </div>
        </div>
          <div>
            <h3>What To Expect</h3>
            <p><em>Everything</em> is about to become a very important word, or at least <em>everything</em> as I see it.</p>
            <p>As you bounce around this new site, you'll notice in the upper-left-hand corners that I change the descriptors under my name more often than a chameleon changes color (though I don't think they change color as often as we like to believe). This is intentional, and not, I hope, a defense mechanism (as is the case with chameleons).</p>
            <p>The topics, ideas, and identities I hint at in those page banners and on the site's landing page are precisely the matters we're going to explore here going forward. We'll talk tech, language, literature, and life, and we'll do it in ways that stimulate our curiosity and expand our understanding of ourselves and the universe we live in.</p>
            <p>That's the goal, anyway.</p>
          </div>
          <div class='Text-And-Image-Container image-on-left'>
            <div class='text-block'>
              <h3>The Philosophy Going Forward</h3>
              <p>To achieve that goal, we won't embrace a regular posting schedule; I want to ensure that what I'm sharing with you is more than something thrown together to meet a personally-imposed deadline.</p>
              <p>In this way, we'll focus more on quality than quantity, on purpose more than frequency.</p>
              <p>And, if some of the pursuits I'm kicking around prove to be worth their while, there might be some surprises in store for you, too.</p>
              <h3>A Work in Progress</h3>
              <p>In the meantime, I ask for your patience. Building this site from the ground up was no done-in-a-day task, and mapping the WordPress version to this new one was not without its tribulations.</p>
              <p>What this means from a visitor's perspective is that, for legacy blog posts, you may encounter awkward formatting or missing links. Images may be strangely sized (or missing altogether), and some previously available content may have been removed so that it can be refined and repurposed.</p>
              <p>I promise you, though, your patience will pay off. It's all for the better, in the end, because we're better when we're all of us—when we're <em>everything people</em>—and I'm so looking forward to the <em>everything</em> we'll discover together.</p>
            </div>
            <div class='image-block'>
              <img src='/favicon-2024-ryan-r-campbell.png' />
              <div class='caption-container'>
                <span>Surprises incoming? Time will tell...</span>
              </div>
            </div>
          </div>
        </div>`,
      protected: false
    },
    excerpt: {
      rendered: "It's been a while—nearly two years, to be exact—but golly, do I have some updates.",
      protected: false
    },
    tags: ['Site Migration', 'Web Development', 'On Writing'],
    episode_featured_image: "/post-content/2024/05/welcome-to-the-new-site-cover-img.png"
  }
];


export default posts
