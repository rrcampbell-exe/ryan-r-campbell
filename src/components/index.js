import BlogTags from './Blog/BlogTags'
import BlogTeaser from './Blog/BlogTeaser'
import Books from './Books/Books'
import Builds from './Builds/Builds'
import Contact from './Contact/Contact'
import Footer from './Footer/Footer'
import Header from './Header/Header'
import PageWrapper from './PageWrapper/PageWrapper'
import SubNav from './SubNav/SubNav'
import Reviews from './Reviews/Reviews'
import NotFound from './NotFound/NotFound'
import NotificationBanner from './NotificationBanner/NotificationBanner'
import Loading from './Loading/Loading'
import Quote from './Quote/Quote'
import TextAndImage from './TextAndImage/TextAndImage'
import HamburgerMenu from './HamburgerMenu/HamburgerMenu'
import PopText from './PopText/PopText'
import PodcastTeaser from './Podcast/PodcastTeaser'
import Boasts from './Tech/Boasts'
import SkillsCloud from './Tech/SkillsCloud'
import BlogFooter from './Blog/BlogFooter'
import SubstackRedirect from './SubstackRedirect/SubstackRedirect'
import { CodeSpan, CodeBlock } from './CodeBlock/CodeBlock'

export { BlogTeaser, Books, Builds, Footer, Header, PageWrapper, SubNav, Contact, Reviews, NotFound, NotificationBanner, Loading, Quote, TextAndImage, BlogTags, HamburgerMenu, PopText, PodcastTeaser, Boasts, SkillsCloud, BlogFooter, CodeSpan, CodeBlock, SubstackRedirect }
