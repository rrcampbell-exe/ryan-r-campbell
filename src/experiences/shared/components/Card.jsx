import React from 'react'
import styled from 'styled-components'

// TODO: size should flex based on number of cards
// TODO: need to set up theme colors

const StyledDiv = styled.div`
  height: 16rem;
  width: 16rem;
  border: 1px solid #242424;
  border-radius: 4px;
  padding: 1rem;
  margin: 1rem;
`

const Card = ({ children }) => {
  return (
    <StyledDiv>
      {children}
    </StyledDiv>
  )
}

export default Card
