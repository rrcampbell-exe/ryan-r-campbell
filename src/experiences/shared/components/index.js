import BlogPost from './Blog/BlogPost'
import BlogTeaser from './Blog/BlogTeaser'
import Books from './Books/Books'
import Builds from './Builds/Builds'
import Footer from './Footer/Footer'
import Header from './Header/Header'
import PageWrapper from './PageWrapper/PageWrapper'
import SubNav from './SubNav/SubNav'
import Contact from './Contact/Contact'
import Reviews from './Reviews/Reviews'
import NotFound from './NotFound/NotFound'
import NotificationBanner from './NotificationBanner/NotificationBanner'

export { BlogPost, BlogTeaser, Books, Builds, Footer, Header, PageWrapper, SubNav, Contact, Reviews, NotFound, NotificationBanner }
