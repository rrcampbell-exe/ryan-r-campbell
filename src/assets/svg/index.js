import { BrowserSVG, LaptopSVG, ManagementSVG } from './BoastsSVG'
import CloseSVG from './CloseSVG'
import ExternalLinkSVG from './ExternalLinkSVG'
import HamburgerIcon from './HamburgerIcon'
import MoonSVG from './MoonSVG'
import QuotesSVG from './QuotesSVG'
import RobotSVG from './RobotSVG'
import SunSVG from './SunSVG'
import CalendarSVG from './CalendarSVG'

export { BrowserSVG, LaptopSVG, ManagementSVG, CloseSVG, ExternalLinkSVG, HamburgerIcon, MoonSVG, QuotesSVG, RobotSVG, SunSVG, CalendarSVG }
