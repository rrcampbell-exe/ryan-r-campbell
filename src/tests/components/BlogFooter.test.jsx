import { describe, expect, test, vi } from 'vitest'
import { BlogFooter } from '../../components'
import { posts, legacyPosts } from '../../assets/posts'
import { render } from '@testing-library/react'
import ContextAndRouterProvider from '../test-utils'

const mockNavigate = vi.fn()

vi.mock('react-router-dom', async (importOriginal) => {
  const actual = await importOriginal()
  return {
    ...actual,
    useNavigate: () => mockNavigate
  }
})

const allPosts = [...posts, ...legacyPosts]

describe('BlogFooter', () => {
  test('default snapshot', () => {
    const blogFooter = render(
      <ContextAndRouterProvider>
        <BlogFooter post={allPosts[1]} />
      </ContextAndRouterProvider>
    )
    expect(blogFooter).toMatchSnapshot()
  })
  test('should render BackToBlog on latest post', () => {
    allPosts[0].postIndex = 0
    const { getByText } = render(
      <ContextAndRouterProvider>
        <BlogFooter post={allPosts[0]} />
      </ContextAndRouterProvider>
    )

    const backToBlog = getByText('back to blog')
    expect(backToBlog).toBeInTheDocument()
  })
  test('should render BackToBlog on oldest post', () => {
    allPosts[allPosts.length - 1].postIndex = allPosts.length - 1
    const { getByText } = render(
      <ContextAndRouterProvider>
        <BlogFooter post={allPosts[allPosts.length - 1]} />
      </ContextAndRouterProvider>
    )

    const backToBlog = getByText('back to blog')
    expect(backToBlog).toBeInTheDocument()
  })
  test('should navigate to previous post', () => {
    const { getByText } = render(
      <ContextAndRouterProvider>
        <BlogFooter post={allPosts[1]} />
      </ContextAndRouterProvider>
    )

    const previousPost = getByText('< previous post')
    const nextPost = getByText('next post >')
    previousPost.click()
    nextPost.click()
    expect(mockNavigate).toHaveBeenCalledTimes(2)
  })
  test('should not display previous post on oldest post', () => {
    allPosts[allPosts.length - 1].postIndex = allPosts.length - 1
    const { queryByText } = render(
      <ContextAndRouterProvider>
        <BlogFooter post={allPosts[allPosts.length - 1]} />
      </ContextAndRouterProvider>
    )

    const previousPost = queryByText('< previous post')
    expect(previousPost).toBeNull()
  })
  test('should not display next post on latest post', () => {
    allPosts[0].postIndex = 0
    const { queryByText } = render(
      <ContextAndRouterProvider>
        <BlogFooter post={allPosts[0]} />
      </ContextAndRouterProvider>
    )

    const nextPost = queryByText('next post >')
    expect(nextPost).toBeNull()
  })
})
