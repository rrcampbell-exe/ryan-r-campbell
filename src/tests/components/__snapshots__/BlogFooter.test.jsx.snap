// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BlogFooter > default snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="nav-footer"
      >
        <div
          class="post-nav-container previous-post"
        >
          <a
            title="link to previous post"
          >
            &lt; previous post
          </a>
          <span />
        </div>
        <div
          class="post-nav-container next-post"
        >
          <a
            title="link to next post"
          >
            next post &gt;
          </a>
          <span />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="nav-footer"
    >
      <div
        class="post-nav-container previous-post"
      >
        <a
          title="link to previous post"
        >
          &lt; previous post
        </a>
        <span />
      </div>
      <div
        class="post-nav-container next-post"
      >
        <a
          title="link to next post"
        >
          next post &gt;
        </a>
        <span />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
